### PR TITLE
[Feature] mock audioplayer

### PR DESCRIPTION
**Pull request by SentulAsia**
Added mock for audio player

### CODE REVIEW

**Comment by SentulAsia** [comment]
@ricardopereira please merge this one first, and I will fix the conflict on the other PR once you've merged this.

File: ColiseuPlayer/ColiseuPlayer.swift [hunk 1]
```diff
@@ -294,13 +307,8 @@ public class ColiseuPlayer: NSObject
...
 
-        do {
-            if let path = song.path {
-                self.audioPlayer = try AVAudioPlayer(contentsOf: path)
-            }
-        }
-        catch let error {
-            print("AVAudioPlayer error occurred:\n \(error)")
+        if let path = song.path {
+            self.audioPlayer = self.engine.initAudioPlayer(url: path)
```
Comment (ricardopereira) [inline ColiseuPlayer/ColiseuPlayer.swift]: I see this like a Builder and not an Engine, agree? The class is building AudioPlayer's.

File: ColiseuPlayerTests/ColiseuPlayerTests.swift [hunk 2]
```diff
@@ -113,7 +114,7 @@ class ColiseuPlayerTests: XCTestCase {
 
     func testPlaySongIndexIsInvalidIndex() {
         // given
-        let expectedResult = true
+        let expectedResult = false
```
Comment (ricardopereira) [inline ColiseuPlayerTests/ColiseuPlayerTests.swift]: Good catch 😄

File: ColiseuPlayer/ColiseuPlayer.swift (hunk 1 above)
Comment (ricardopereira) [reply ColiseuPlayer/ColiseuPlayer.swift]: I see this more like `self.builder.createAudioPlayer`.

File: ColiseuPlayerTests/ColiseuPlayerTests.swift (hunk 2 above)
Comment (SentulAsia) [reply ColiseuPlayerTests/ColiseuPlayerTests.swift]: yeah... when checking the test then I realized that I've overlooked on this...

File: ColiseuPlayer/ColiseuPlayer.swift (hunk 1 above)
Comment (SentulAsia) [reply ColiseuPlayer/ColiseuPlayer.swift]: Yeah, you are correct...

**Comment by ricardopereira** [comment]
Perfect, thanks!

**Pull request merged**